packages/fast-figma-plugin-msft/src/messaging/common.ts

```diff
@@ -0,0 +1,44 @@
+import { isString } from "lodash-es";
+/**
+ * There are two general types of messages, plugin ui -> Figma document and Figma document -> plugin ui.
+ * In this application, events originating from the plugin ui are referred to as UI_EVENTs
+ * while events originating in the Figma document are referred to as CANVAS_EVENTs
+ *
+ * All messages sent by both the canvas and UI implement the BaseEventData structure.
+ */
+
+export interface PluginMessageData<T extends string> {
+    /**
+     * A message sent by either the UI or the Canvas will always contain
+     * a type field that is a symbol
+     */
+    type: T;
+}
+
+// Assert the MessageEvent is a PluginMessageEvent
+export function isPluginMessageEvent(
+    e: MessageEvent | PluginMessageEvent
+): e is PluginMessageEvent {
+    try {
+        return !!(
+            e &&
+            e.data &&
+            isString(e.data.pluginMessage) &&
+            isString(e.data.pluginId) &&
+            JSON.parse(e.data.pluginMessage)
+        );
+    } catch (e) {
+        return false;
+    }
+}
+
+/**
+ * Expand uppon the browser's MessageEvent object to define
```

Comment:
nit: upon

Also, not too sure what you're trying to say here, did you mean "Expand on the browser's MessageEvent object to define Figma plugin data"?

Reply:
I'll re-phrase - just means that the "data" property in Figma follows a strict format that isn't enforced by the `MessageEvent` type - should probably be called a `FigmaMessageEvent`. I'll also see if they export something from their d.ts file for this, cause they *should*